src/bootstrap/check.rs

```diff
@@ -106,7 +106,8 @@ pub fn cargo(build: &Build, stage: u32, host: &str) {
     let ref newpath = format!("{}{}{}", path.display(), sep, old_path);
 
     let mut cargo = build.cargo(compiler, Mode::Tool, host, "test");
-    cargo.arg("--manifest-path").arg(build.src.join("src/tools/cargo/Cargo.toml"));
+    cargo.arg("--manifest-path").arg(build.src.join("src/Cargo.toml"));
+    cargo.arg("--project").arg("cargo");
```

Comment:
Ok, can this be backed out then? I think this has clearer intent that Cargo's being built (pointing at Cargo's manifest)